test_conformance/vulkan/test_vulkan_interop_buffer.cpp

```diff
@@ -229,14 +251,37 @@ int run_test_with_two_queue(cl_context &context, cl_command_queue &cmd_queue1,
 
                 if (iter == 0)
                 {
-                    vkQueue.submit(vkCommandBuffer, vkVk2CLSemaphore);
+                    if (use_fence)
+                    {
+                        vkQueue.submit(vkCommandBuffer, fence);
+                    }
+                    else
+                    {
+                        vkQueue.submit(vkCommandBuffer, vkVk2CLSemaphore);
+                    }
+                }
+                else
+                {
+                    if (use_fence)
+                    {
```

Comment:
Fence needs to be reset before reusing after previous Wait.

Reply:
I've added it but now there are deadlocks there.

Reply:
The sequence should be "Reset fence, use it in some submit operation where it gets signaled, wait on it" or  "use fence in some submit operation where it gets signaled, wait on it, Reset fence". Resetting just before wait but after signal would override effect of signalling at the end of completion of work submitted earlier and wait would not wait for the same.

Reply:
After this change there is an issue with:

```
err = clEnqueueReadBuffer(cmd_queue1, error_1, CL_TRUE, 0,
                                              sizeof(uint8_t), error_3, 0, NULL,
                                              NULL);
```

it produces the following output:

```
log_error("&&&& vulkan_opencl_buffer test FAILED\n");
```

Yet I am not sure what the reason is.

Reply:
Discussed May 9th.  @nikhiljnv will try to reproduce these failures.

Reply:
We tried repro of the issue and found that synchronizations are misplaced in some scenarios.
Have requested changes which should fix data mismatches seen here.

Reply:
The reason for the else branch when iter > 0 was to wait on the semaphore dispatched at the end of the last frame.  We either need to wait on the last fence here, or wait on a fence at the end of the frame, and delete the if(iter==0) behavior.